src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ReferralService.kt

```diff
@@ -26,7 +31,10 @@ class ReferralService(val repository: ReferralRepository, val authUserRepository
 
     // fixme: hardcoded for now
     referral.referenceNumber = "HDJ2123F"
-    return repository.save(referral)
+
+    val sentReferral = repository.save(referral)
+    eventPublisher.referralSentEvent(sentReferral)
```

Comment:
I'm wondering -- do we benefit from the `ReferralEventPublisher` abstraction?

It could also be
```kt
return repository.save(referral)
   .also { applicationEventPublisher.publishEvent(ReferralSentEvent(this, sentReferral)) }
```

Reply:
`ApplicationEventPublisher` is in interface only.